kin-devplatform-sdk/src/main/java/kin/devplatform/data/blockchain/BlockchainSource.java

```diff
@@ -46,24 +46,22 @@ void sendTransaction(@NonNull String publicAddress, @NonNull BigDecimal amount,
 	void getBalance(@Nullable final KinCallback<Balance> callback);
 
 	/**
-	 * Add balance observer in order to start receive balance updates
+	 * Reconnect the balance connection, due to connection lose.
 	 */
-	void addBalanceObserver(@NonNull final Observer<Balance> observer);
+	void reconnectBalanceConnection();
 
 	/**
-	 * Add balance observer that will keep a connection on account balance updates from the blockchain network.
+	 * Add balance observer in order to start receive balance updates
+	 *
+	 * @param startSSE true will keep a connection on account balance updates from the blockchain network
```

Comment:
And if false then when connection lost it will not try to reconnect? and how can we know that the connection has been lost? do we have a listener for that?

Reply:
there's a possibility here to listen to "internal" balance updates, meaning whenever someone asks for balance directly (`getBalance`) without SSE, the module will notify any observer.
the `true` is indicates actually that we want to listen directly to the blockchain.
Having said that I think there's no need for that anymore, and we should use SSE always, but didn't want to make deep changes now.